drivers/platform/aducm3029/iio_aducm3029.c

```diff
@@ -0,0 +1,371 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "pwm.h"
+#include "util.h"
+#include "gpio.h"
+#include "iio_aducm3029.h"
+#include "aducm3029_adc.h"
+#include "error.h"
+#include <stdio.h>
+#include <inttypes.h>
+#include <sys/platform.h>
+
+#define GPIO_PIN(nb)	(1u << ((nb) & 0x0F))
+#define GPIO_PORT(nb)	(((nb) & 0xF0) >> 4)
+
+volatile uint32_t *pinmux_addrs[] = {
+	((volatile uint32_t *)REG_GPIO0_CFG),
+	((volatile uint32_t *)REG_GPIO1_CFG),
+	((volatile uint32_t *)REG_GPIO2_CFG)
+};
+
+static struct adc_init_param default_adc_init_param = {0};
+static struct pwm_init_param default_pwm_init_par = {
+	.id = 0,
+	.duty_cycle_ns = 5000000,
+	.period_ns = 10000000,
+	.polarity = PWM_POLARITY_HIGH,
+	.extra = NULL
+};
+static struct gpio_init_param default_gpio_init_par = {
+	.number = 0,
+	.extra = NULL
+};
+
+/*
+ * adc_mux[0][0] -> ADC0 port,
+ * adc_mux[0][1] -> ADC0 pin,
+ * adc_mux[0][2] -> mux_val
+ */
+static uint32_t adc_muxs[ADUCM3029_ADC_NUM_CH][3] = {
+	{2, 3, 1},
+	{2, 4, 1},
+	{2, 5, 1},
+	{2, 6, 1},
+	{2, 7, 1},
+	{2, 8, 1}
+};
+
+static uint32_t timers_muxs[ADUCM3029_TIMERS_NUMS][3] = {
+	{0, 14, 1},
+	{1, 11, 2},
+	{2, 1, 2}
+};
+
+enum pin_type {
+	PIN_TYPE_GPIO,
+	PIN_TYPE_ADC,
+	PIN_TYPE_TIMER
+};
+
+/* Enable in pin mux the needed type */
+static void set_pin(uint32_t id, enum pin_type type)
+{
+	volatile uint32_t *port_addr;
```

Comment:
Where is this variable used?

Reply:
It is the address of an internal register